lib-ui-compose/src/commonMain/kotlin/com/ustadmobile/libuicompose/view/report/edit/ReportEditScreen.kt

```diff
@@ -0,0 +1,355 @@
+                }
+            }
+
+            // Filters Section
+            item {
+                if (!seriesItem.reportSeriesFilters.isNullOrEmpty()) {
+                    Text(
+                        text = stringResource(MR.strings.filters),
+                        modifier = Modifier.defaultScreenPadding()
+                    )
+                }
+                seriesItem.reportSeriesFilters?.forEachIndexed { index, reportFilter2 ->
+                    Row(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .defaultScreenPadding()
+
+                    ) {
+                        val fieldName = reportFilter2.reportFilterField?.name?.lowercase()
+                            ?.replaceFirstChar { it.uppercase() } ?: ""
+                        val comparisonSymbol =
+                            reportFilter2.reportFilterCondition?.symbol ?: ""
+                        val filterText =
+                            "$fieldName $comparisonSymbol ${reportFilter2.reportFilterValue}"
+
+                        Text(
+                            text = filterText,
+                            modifier = Modifier.weight(1f)
+                        )
+                        Icon(
+                            imageVector = Icons.Filled.Close,
+                            contentDescription = "Remove filter",
+                            modifier = Modifier
+                                .clickable {
+                                    onRemoveFilter(index, seriesItem.reportSeriesUid)
+                                }
+                        )
+                    }
+                }
+            }
+
+            item {
+                Button(
+                    onClick = { onAddFilter(seriesItem.reportSeriesUid) },
+                    modifier = Modifier.fillMaxWidth().defaultScreenPadding()
+                ) {
+                    Text(
+                        text = stringResource(MR.strings.add_filter),
+                    )
+                }
+            }
+        }
+
+        item {
+            Button(onClick = { onAddSeries() }, modifier = Modifier.fillMaxWidth()) {
+                Text(
+                    text = stringResource(MR.strings.add_series),
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun LabeledDropdownMenu(
+    label: String,
+    options: List<MessageIdOption2>,
+    selectedValue: Int,
+    onOptionSelected: (MessageIdOption2) -> Unit
+){
+    Column(
+        modifier = Modifier.defaultScreenPadding(),
+        verticalArrangement = Arrangement.spacedBy(8.dp)
+
+    ) {
+        Text(text = label, maxLines = 1)
+        ExposedDropdownMenu(
+            options = options,
+            selectedValue = selectedValue,
+            onOptionSelected = onOptionSelected
+        )
+    }
+}
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun ExposedDropdownMenu(
+    options: List<MessageIdOption2>,
```

Comment:
This shouldn't need to use MessageIdOption2. 

I recommend:
1) Create an interface e.g. OptionWithLabel
```
interface OptionWithLabelStringResource {
    val label: StringResource
}
```

2. Update the enum classes to implement the interface above

3. the ExposedDropDownMenu function will then be much easier/cleaner e.g.

```
fun <T: OptionWithLabelStringResource> ExposedDropdownMenu(
   options: List<T>,
   selectedValue: T,
   onOptionSelected: (T) -> Unit
)
```

Then there will be no need to map to/from integers etc.